### ISSUE TITLE
about the RotaryEmbedding

### ISSUE THREAD

**Issue by tricky61** [vllm-project/vllm#5984]

**Comment by tricky61** [comment]
and I dont't know why use repeat_interleave. How to modify the kernel when I want to use the rotary_emb like this:
https://github.com/JunnYu/RoFormer_pytorch

def apply_rotary(x, sinusoidal_pos=None):
    if sinusoidal_pos is None:
        return x
    sin, cos = sinusoidal_pos
    # x.shape [batch, seq_len, 2]
    x1, x2 = x[..., 0::2], x[..., 1::2]
    return torch.cat([x1 * cos - x2 * sin, x1 * sin + x2 * cos], dim=-1)
